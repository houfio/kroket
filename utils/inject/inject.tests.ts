import { inject } from './index';

const style = 'body { background-color: dodgerblue; }';

afterEach(() => {
  const element = document.querySelector('[data-kroket]');

  element?.parentNode?.removeChild(element);
});

it('should add a style tag to the head', () => {
  inject(true);

  expect(document.head).toMatchSnapshot();
});

it('should add a text node to the style tag', () => {
  inject(true)([style]);

  expect(document.head).toMatchSnapshot();
});

it('should insert a rule to the style tag', () => {
  const execute = inject(false);

  execute([style]);

  expect(document.head).toMatchSnapshot();
  expect(execute.sheet.cssRules.length).toBe(1);
});
