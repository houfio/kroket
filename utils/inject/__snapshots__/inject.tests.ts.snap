// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a style tag to the head 1`] = `
<head>
  <style
    data-kroket=""
  />
</head>
`;

exports[`should add a text node to the style tag 1`] = `
<head>
  <style
    data-kroket=""
  >
    body { background-color: dodgerblue; }
  </style>
</head>
`;

exports[`should insert a rule to the style tag 1`] = `
<head>
  <style
    data-kroket=""
  />
</head>
`;
